startstoptorq.sh

```diff
@@ -0,0 +1,275 @@
+      echo `date '+%H:%M:%S'` "| $i failed - unavailable processname"
+    fi
+  done
+  procs=$avail                                                                                      # assign valid processes
+ }
+
+getprocs() {
+  if [ "$2" == "all" ]; then 
+    procs=$(getall)                                                                                 # get all processes
+  else
+    shift                                                                                           # ignore first argument
+    checkinput $@                                                                                   # check input process names
+  fi
+ }
+
+flag() {
+  count=0
+  for i in ${BASH_ARGV[*]};
+  do
+    count=$(($count+1))
+    if [[ $i == "-$1" ]]; then                                                                      # find flag argument
+      n=$(($count-2))
+    fi
+  done
+ }
+
+flagextras() {
+  flag $@
+  z=$n
+  while [ $z -ge 0 ]
+  do
+    extras+="${BASH_ARGV[$z]} "                                                                     # get all extra parameters 
+    z=$[$z-1]
+  done
+ }
+
+flagcsv() {
+  flag $@
+  csvpath="${BASH_ARGV[$n]}"                                                                        # assign specifed csv file
+ }
+
+getextras() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]]; then                                            
+    eval flagextras "extras";
+    length=$(($#-$n-2));
+    array=${@:1:$length};                                                                           # arguments without extras flag
+  else
+    array=$@;
+  fi
+ }
+
+getcsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    eval flagcsv "csv";
+    length=$(($#-2));
+    array=${@:1:$length};                                                                           # arguments without csv flag
+    getprocs $array;
+  else
+    csvpath=$DEFAULTCSV;                                                                            # set csv file to default
+    getprocs $array;
+  fi
+ }
+
+getextrascsv() {
+  eval flagextras "extras";
+  eval flagcsv "csv";
+  length=$(($#-$n-2));
+  array=${@:1:$length};                                                                             # arguments without extras and csv flag
+  getprocs $array;
+ }
+
+checkextrascsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]] && [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    getextrascsv $@;
+  else
+    getextras $@;
+    getcsv $@;
+  fi
+ }
+
+allcsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    eval flagcsv "csv";
+  else
+    csvpath=$DEFAULTCSV;                                                                            
+  fi
+ }
+
+startprocs() {
+  for p in $procs;
+  do
+    start $p;
+  done
+ }
+
+stopprocs() {
+  for p in $procs;
+  do
+    stop $p;
+  done
+ }
+
+usage() {
+  printf -- "Arguments:\n"
+  printf -- "-procs                               to list all processes\n"
```

Comment:
I think maybe some additional indentation on these strings would be good, to separate from section headers